### PR TITLE
shield values examine

### PR DESCRIPTION
**Pull request by Slava0135**
## About the PR

**Media**

Peek 2023-06-10 00-04

**Changelog**

:cl:
- add: Shield values now can be examined

### CODE REVIEW

**Comment by Slava0135** [comment]
image
changed it a bit

File: Content.Server/Blocking/BlockingSystem.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    private void OnVerbExamine(EntityUid uid, BlockingComponent component, GetVerbsEvent<ExamineVerb> args)
+    {
+        if (!args.CanInteract || !args.CanAccess)
+            return;
+
+        _proto.TryIndex<DamageModifierSetPrototype>(component.PassiveBlockDamageModifer, out var passiveblockModifier);
+        _proto.TryIndex<DamageModifierSetPrototype>(component.ActiveBlockDamageModifier, out var activeBlockModifier);
+
+        var fraction = component.IsBlocking ? component.ActiveBlockFraction : component.PassiveBlockFraction;
+        var modifier = component.IsBlocking ? activeBlockModifier : passiveblockModifier;
```
Comment (tom-leys) [inline Content.Server/Blocking/BlockingSystem.cs:32]: You should decide which of these prototypes you need prior to calling `_proto.TryIndex` to avoid doing two lookups when you only want one.

**Review by tom-leys — APPROVED** [review]
LGTM, just a simple performance issue doing two lookups.

**Comment by keronshb** [comment]
I will review this sometime within the week.

File: Content.Server/Blocking/BlockingSystem.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
+using Content.Server.Examine;
+using Content.Shared.Blocking;
+using Content.Shared.Damage;
+using Content.Shared.Damage.Prototypes;
+using Content.Shared.Verbs;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Utility;
+
+namespace Content.Server.Blocking;
```
Comment (keronshb) [inline Content.Server/Blocking/BlockingSystem.cs:9]: I don't see the need to make a server class. You can access GetVerbsEvent from shared. Please revert this change and add what you did here into Shared instead.

File: Content.Shared/Blocking/BlockingSystem.User.cs [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 namespace Content.Shared.Blocking;
 
-public sealed partial class BlockingSystem
+public abstract partial class SharedBlockingSystem
```
Comment (keronshb) [inline Content.Shared/Blocking/BlockingSystem.User.cs:8]:
```suggestion
public sealed partial class BlockingSystem
```
Revert this back to BlockingSystem

File: Content.Shared/Blocking/Components/BlockingUserComponent.cs [hunk 4]
```diff
@@ -4,7 +4,7 @@
 namespace Content.Shared.Blocking;
 
 /// <summary>
-/// This component gets dynamically added to an Entity via the <see cref="BlockingSystem"/>
+/// This component gets dynamically added to an Entity via the <see cref="SharedBlockingSystem"/>
```
Comment (keronshb) [inline Content.Shared/Blocking/Components/BlockingUserComponent.cs:7]: Same here

File: Resources/Locale/en-US/blocking/blocking-examine.ftl [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# Blocking examines
+blocking-examinable-verb-text = Blocking
```
Comment (keronshb) [inline Resources/Locale/en-US/blocking/blocking-examine.ftl:2]: You should rename this to something else. Maybe Stats? Or Item Stats?

File: Content.Server/Blocking/BlockingSystem.cs [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+        SubscribeLocalEvent<BlockingComponent, GetVerbsEvent<ExamineVerb>>(OnVerbExamine);
+    }
+
+    private void OnVerbExamine(EntityUid uid, BlockingComponent component, GetVerbsEvent<ExamineVerb> args)
+    {
+        if (!args.CanInteract || !args.CanAccess)
+            return;
+
+        _proto.TryIndex<DamageModifierSetPrototype>(component.PassiveBlockDamageModifer, out var passiveblockModifier);
+        _proto.TryIndex<DamageModifierSetPrototype>(component.ActiveBlockDamageModifier, out var activeBlockModifier);
```
Comment (keronshb) [inline Content.Server/Blocking/BlockingSystem.cs:29]:
```suggestion
    var activeBlockModifier = component.ActiveBlockDamageModifier;
    var passiveblockModifier = component.PassiveBlockDamageModifer;
```

Instead of calling _proto.TryIndex you should be able to do this change and then make modifications to the review below. If you need an example check out Server.ArmorSystem line 70.

**Review by keronshb — CHANGES_REQUESTED** [review]
